File: lists/frame.py


def frame(arr):
    max_length = max(len(i) for i in arr)
    total_length = max_length + 4

    border = ''
    for x in range(total_length):
        border += '*'

    print(border)
    for a in arr:
        spaces = max_length - len(a)
        temp = '* ' + a + ' ' * spaces + ' *'
        print(temp)
    print(border)
    


arr = ['Hello', 'World', 'See', 'More', 'Words']

frame(arr)

File: lists/frame.js


const frame = arr => {
    let lengths = arr.map(a => a.length);
    let max_length = Math.max(...lengths);
    let total_length = max_length + 4;

    let border = '*'.repeat(total_length);

    console.log(border);
    for (let x in arr) {
        let spaces = max_length - arr[x].length;
        let temp = '* ' + arr[x] + ' '.repeat(spaces) + ' *';
        console.log(temp);
    }
    console.log(border);
}


let arr = ['Hello', 'World', 'one', 'more', 'time'];

frame(arr);